frontend/server/fileSystem.js

```diff
@@ -92,8 +90,8 @@ export async function filterDirectories(filePaths) {
   const stats = await Promise.all(filePaths.map((p) => fs.stat(p)));
   return filePaths
     .map((p, i) => [p, stats[i]])
-    .filter(([p, s]) => s.isDirectory())
-    .map(([p, s]) => p);
+    .filter(([, s]) => s.isDirectory())
```

Comment:
i don't know if the linter complains about this line and the next line but it needs to stay how it was before (maybe with more descriptive names?) the filter operation should return the full pair, as in the last map operation it's returning the final `p` (path) if `s.isDirectory()`

a way to write this more understandably would be

```
.filter(([filePath, stat]) => s.isDirectory())
.map(([filePath, stat]) => filePath)
```
The code is taking advantage of es6 implicit return syntax for functions

Reply:
For this one the linter was complaining about p and s being unused. So we're ignoring them in the respective lambdas

Reply:
Maybe I'm misunderstanding the syntax, but if the filter returns the two object array then mapping that to a single element `[p]` would change the output of the function

Reply:
We must be misunderstanding each other, because I'm not sure if I understand your point. To me, it seems like these two versions are equivalent. It's taking advantage of the destructuring assignment on the array.

![image](https://github.com/zetane/ZetaForge/assets/35940434/164b584d-7bbc-4f04-8e49-bbad42b3d9e5)

Reply:
ok, i'm wrong, all good